### ISSUE TITLE
Name ' ' cannot be defined pycharm

### ISSUE THREAD

**Issue by ahmedmsoliman96** [Cartucho/mAP#74]
https://github.com/Cartucho/mAP/blob/20d2f89d90a1b5ca119871e925132af32ab61b74/main.py#L396
       
 bbox = left + " " + top + " " + right + " " +bottom
will this line work since left top right bottom variables are defined in another scope ?

**Comment by Cartucho** [comment]
what do you mean by another scope?

Those variables are being read from a line of the file:               https://github.com/Cartucho/mAP/blob/20d2f89d90a1b5ca119871e925132af32ab61b74/main.py#L382

or

https://github.com/Cartucho/mAP/blob/20d2f89d90a1b5ca119871e925132af32ab61b74/main.py#L385

**Comment by ahmedmsoliman96** [comment]
i am not expert in python but i guess . but for example in java  a variable or object declared inside a try catch clause or a method or any other that goes inside {} is not visible to other parts of the class . that's why i was confused that will this work in pyhton  or not 
also when i tried to paste the code in pycharm i get this msg 

![image](https://user-images.githubusercontent.com/27621581/57642790-75a9b800-75b8-11e9-8298-5c45610651e4.png)

however the code run and does not produce any error so i guess in python variables that are initialized in try except are accessible  else where .

**Comment by Cartucho** [comment]
Yes, they are accessible. In this case, if it is not able to read it will throw an error.
Anyway, you are right that the variables should be initialized before.

**Comment by Bekithemb** [comment]
> i am not expert in python . but for example in java a variable or object declared inside a try catch clause or a method or any other that goes inside {} is not visible to other parts of the class . that's why i was confused that will this work in pyhton or not also when i tried to paste the code in pycharm i get this msg
> 
> ![image](https://user-images.githubusercontent.com/27621581/57642790-75a9b800-75b8-11e9-8298-5c45610651e4.png)
> 
> however the code run and does not produce any error so i guess in python variables that are initialized in try except are accessible else where .

how do i correct this
